### ISSUE TITLE
Lua bridge to Rust

### ISSUE THREAD

**Issue by kazimir-malevich** [tag1consulting/goose#347]
What would be your opinions on using a Lua bridge to Rust in order to  write the load test scripts in Lua This I feel may gain some traction to the tool, is this is instead possible?

**Comment by jeremyandrews** [comment]
This is not on my personal roadmap, but I'd certainly accept PRs working toward this as an option.

**Comment by kazimir-malevich** [comment]
There is some options here, and we'd probably want a switch to code in Rust or another **mainstream** dynamic language. 
https://arewegameyet.rs/ecosystem/scripting/
https://www.hobofan.com/rust-interop/
https://areweextendingyet.github.io/

**Comment by kazimir-malevich** [comment]
[rhai](https://crates.io/crates/rhai) has a major version and is "similar to JavaScript+Rust"

**Comment by kazimir-malevich** [comment]
I am having second thoughts about this and thinking of closing. The Rust compiler and rust-analyzer are extremely compelling. But the GH stars should not be in the hundreds, but thousands as in my opinion this is the best and most comprehensive load tool out there. WDYT @jeremyandrews? Maybe you are just too far in the future!!!

**Comment by jeremyandrews** [comment]
I also agree that the Rust tools are compelling, but I think we should leave this issue open as it's a frequent comment/request that we should support a simpler language for writing tests. While it's still not on my personal roadmap, I'd not be opposed if someone contributed this as an optional Goose feature.

**Comment by kazimir-malevich** [comment]
OK. I'll look into this.

**Comment by kazimir-malevich** [comment]
Maybe [AssemblyScript](https://www.assemblyscript.org/) with WASM. This is not going to be trivial.

**Comment by jeremyandrews** [comment]
That certainly sounds interesting. It would be ideal to explore adding this as a separate project with a Goose dependency -- as ultimately it could be nice to support more than one embedded dynamic language. (Similar to how the goal is to split out Reqwest and to support multiple rust HTTP clients, as well as other protocols.)
